src/Ocean/HydrostaticBoussinesq/HydrostaticBoussinesqModel.jl

```diff
@@ -676,9 +665,12 @@ function update_auxiliary_state_gradient!(
 )
     FT = eltype(Q)
     A = dg.state_auxiliary
+    D = dg.state_gradient_flux
 
-    f! = nodal_update_auxiliary_state!
-    update_auxiliary_state!(f!, dg, m, Q, t, elems; diffusive = true)
+    # load -∇ʰu as ∂ᶻw
+    if elems == dg.grid.topology.realelems
+        A.w .= -D.∇ʰu
+    end
```

Comment:
How is `A.w` used? Is it used in the flux or just the volume? Since the auxiliary is never comm'd, you will need to update the ghost too.

You may need to use `varsindex` to get the indices and broadcast from the data instead.